### ISSUE TITLE
NIO 2.0 / Vapor 4 support

### ISSUE THREAD

**Issue by ffried** [sersoft-gmbh/swift-smtp#2]
With Vapor 4 coming up, we should add support for it, which also requires supporting NIO 2.0.

**Comment by ffried** [comment]
Just released a first RC for version 2.0: https://github.com/sersoft-gmbh/SwiftSMTP/releases/tag/2.0.0-rc.1.0

**Comment by ffried** [comment]
The RC-releases can be tracked by specifying the following SPM dependency:
```swift
.package(url: "https://github.com/sersoft-gmbh/SwiftSMTP.git", from: "2.0.0-rc"),
```

**Comment by christopherweems** [comment]
Great news, I was able to bring in 2.0.0-rc, build, run, and send email on my Mac and an Ubuntu VM without issues! I believe the extra dependencies are causing the build to choke on my ram-deprived server, but that has little to do with the correctness of the package. I will revisit once I can move compilation off the server in my build chain. Great work! The move to allowing arbitrary loggers via SMTPLogger is something I am very happy to see.

**Comment by ffried** [comment]
Hmm, the core part of SwiftSMTP shouldn't have gained any additional dependencies. And with Swift 5.2, SPM should only get the necessary dependencies. At least that's the theory. 😆

**Comment by ffried** [comment]
Ok, I just double checked this. Seems like `swift package resolve` will still fetch all the additional dependencies (vapor in this case) for some reason. However, `swift build -c release` really only builds what is linked.

Is `swift package update` or `swift package resolve` already failing for you?

**Comment by christopherweems** [comment]
`swift build -c release` still resolves the dependencies for me, but as far as I can tell, they are not actually linked in this case, or at lease some analysis isn't performed that would be for a debug build, because it builds on my low-end server, where the debug build did not. I did a couple of builds from scratch on this server, and it worked both time. It was very helpful of you to investigate, thank you very much!

**Comment by ffried** [comment]
Yea, the resolving still happens (did so for me as well - don't know why, though - could be a bug). But it then doesn't actually _build_ (and thus neither link) the unused dependencies.

**Comment by ffried** [comment]
Quick heads-up: The new RC2 contains a few breaking changes in the core layer. No features were lost, though. Just a parameter that moved into the `Configuration` and a typo fixed.

See the [release notes](https://github.com/sersoft-gmbh/SwiftSMTP/releases/tag/2.0.0-rc.2.0) for more details.

**Comment by ffried** [comment]
This has been released with version 2.0.0
